src/main/java/cz/muni/fi/pv286/mmt/representation/ArrayParser.java

```diff
@@ -0,0 +1,314 @@
+        };
+        stack.add(resultTree);
+
+        while (!stack.isEmpty()) {
+            ResultTree top = stack.get(stack.size() - 1);
+            if (top.getChildIndex() <= top.getChildren().size() - 1) { // Done for each one
+                ResultTree child = top.getChildren().get(top.getChildIndex());
+
+                if (top.getChildIndex() != 0) { // If not first
+                    bytesList.add((byte) ',');
+                    bytesList.add((byte) ' ');
+                }
+
+                if (!child.getChildren().isEmpty() || child.getValue() == null) {
+                    bytesList.add(openingBracket);
+                    stack.add(child);
+                } else if (child.getValue() != null && child.getValue().length >= 1) {
+                    byte[] parsedValue = parseFromByte(child.getValue()[0], outputOption);
+                    for (byte b : parsedValue) {
+                        bytesList.add(b);
+                    }
+                }
+
+                top.setChildIndex(top.getChildIndex() + 1);
+
+            } else {
+                stack.remove(stack.size() - 1);
+                bytesList.add(closingBracket);
+            }
+        }
+
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        bytesList.stream().mapToInt(Byte::intValue).forEach(byteArrayOutputStream::write);
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    private ResultTree parse(byte[] bytes) throws IOException {
+        List<ResultTree> stack = new LinkedList<>();
+        stack.add(new ResultTree(new String(bytes, StandardCharsets.UTF_8)));
+        while (!stack.isEmpty()) {
+            ResultTree current = stack.remove(stack.size() - 1);
+            if (current.getIntermediate() != null) {
+                String extracted = null;
+                Matcher curly = Patterns.curlyBracketsExtractor.matcher(current.getIntermediate());
+                Matcher square = Patterns.squareBracketsExtractor.matcher(current.getIntermediate());
+                Matcher regular = Patterns.regularBracketsExtractor.matcher(current.getIntermediate());
+                if (curly.find()) {
+                    extracted = curly.group(1);
+                } else if (square.find()) {
+                    extracted = square.group(1);
+                } else if (regular.find()) {
+                    extracted = regular.group(1);
+                }
+                if ("".equals(extracted)) {
+                    current.setIntermediate(null);
+                    current.setValue(null);
+                    stack.add(current);
+                } else if (extracted != null) {
+                    stack.add(current);
+                    List<String> tokens = topLevelSplitTokens(extracted);
+                    for (String token : tokens) {
+                        ResultTree t = new ResultTree(token.trim());
+                        current.getChildren().add(t);
```

Comment:
**issue:** Modifying private attribute in another class

Making an attribute private does not make sense when you allow modifying its elements from the outside.
**suggestion:** In class ResultTree create a public addChild method and use this instead of direct modification.